DEPRECATION.md

```diff
@@ -0,0 +1,101 @@
+# Deprecation process
+
+## Introduction
+
+Sometimes it is necessary to remove definitions from `base`, for example because
+they expose GHC-internal implementation details that belong in `ghc-internal` or
+`ghc-experimental` (see [GHC's base libraries: Combining stability with
+innovation](https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/051-ghc-base-libraries.rst)
+for background on these packages and the broader context).
+
+In order to save time and effort for both the CLC and GHC developers, this
+document describes a streamlined standard workflow for deprecation and removal
+of definitions. (This document refers to "definitions" for brevity, but is
+intended to apply to whole modules and re-exports of existing definitions as
+well.)  This functions as an alternative to the [main proposals
+process](./PROPOSALS.md), intended for cases where the definitions being removed
+are manifestly GHC-internal, have a clearly preferable replacement or are
+otherwise unlikely to be controversial. The normal process should be used
+instead for changes that require additional justification (e.g. deprecating a
+widely-used function in a module outside the `GHC.*` namespace).
+
+Proposals may be batched, e.g. by:
+ - domain (e.g. all of `GHC.Stack.*`);
+ - Hackage users (few Hackage users indicates it's easier to vote yes);
+ - deprecation period;
+ - how "internal" they are perceived.
+
+If a batch proposal fails, it's fine to split it into multiple proposals. In
+general, one proposal per module would seem too fine a granularity.
+
+## Process details
+
+The key idea of this process is to separate the decision-making stage (for which
+CLC oversight is required) from the implementation stage (which happens over
+time, without need for further input from the CLC).
```

Comment:
I'm not really sure about it giving infinite leeway. If a proposal to deprecate internals is approved but not acted upon for a long period, to-be-deprecated modules could gain users in the meantime and invalidate the reasoning behind their deprecation.